test/logflare_web/controllers/api/team_controller_test.exs

```diff
@@ -0,0 +1,171 @@
+    test "returns a single team given user and team token", %{
+      conn: conn,
+      user: user,
+      main_team: main_team
+    } do
+      response =
+        conn
+        |> login_user(user)
+        |> get("/api/teams/#{main_team.token}")
+        |> json_response(200)
+
+      assert response["name"] == main_team.name
+    end
+
+    test "returns a single team given user and team token where his not an owner but a member", %{
+      conn: conn,
+      user: user,
+      non_owner_team: non_owner_team
+    } do
+      response =
+        conn
+        |> login_user(user)
+        |> get("/api/teams/#{non_owner_team.token}")
+        |> json_response(200)
+
+      assert response["name"] == non_owner_team.name
+    end
+
+    test "returns not found if doesn't own the team or isn't part of it", %{
+      conn: conn,
+      main_team: main_team,
+      non_owner_team: non_owner_team
+    } do
+      invalid_user = insert(:user)
+
+      conn
+      |> login_user(invalid_user)
+      |> get("/api/teams/#{main_team.token}")
+      |> response(404)
+
+      conn
+      |> login_user(invalid_user)
+      |> get("/api/teams/#{non_owner_team.token}")
+      |> response(404)
+    end
+  end
+
+  describe "create/2" do
+    test "creates a new team for an authenticated user", %{conn: conn} do
+      user = insert(:user)
+      name = TestUtils.random_string()
+
+      response =
+        conn
+        |> login_user(user)
+        |> post("/api/teams", %{name: name})
+        |> json_response(201)
+
+      assert response["name"] == name
+    end
+  end
+
+  describe "update/2" do
+    test "updates an existing team from a user", %{
+      conn: conn,
+      user: user,
+      main_team: main_team
+    } do
+      name = TestUtils.random_string()
+
+      response =
+        conn
+        |> login_user(user)
+        |> patch("/api/teams/#{main_team.token}", %{name: name})
+        |> json_response(204)
+
+      assert response["name"] == name
+    end
+
+    test "returns not found if doesn't own the team", %{conn: conn, main_team: main_team} do
+      invalid_user = insert(:user)
+
+      conn
+      |> login_user(invalid_user)
+      |> patch("/api/teams/#{main_team.token}", %{name: TestUtils.random_string()})
+      |> response(404)
+    end
+  end
+
+  describe "delete/2" do
+    test "deletes an existing team from a user", %{
+      conn: conn,
+      user: user,
+      main_team: main_team
+    } do
+      assert conn
+             |> login_user(user)
+             |> delete("/api/teams/#{main_team.token}")
+             |> response(204)
+    end
+
+    test "returns not found if doesn't own the team", %{conn: conn, main_team: main_team} do
+      invalid_user = insert(:user)
+
+      assert conn
+             |> login_user(invalid_user)
+             |> delete("/api/teams/#{main_team.token}")
+             |> response(404)
+    end
+  end
+
+  test "changeset errors handled gracefully", %{conn: conn, user: user, main_team: main_team} do
```

Comment:
can be moved to the post/patch test cases since this is expected behaviour, will be less verbose.